server/bin/pbench-index.py

```diff
@@ -181,344 +135,15 @@ def main(options, name):
-                        tb, Path(controller_path, linkerrdest), idxctx.logger
-                    )
-                idxctx.logger.info("Finished {} (size {:d})", tb, size)
-        except Exception:
-            idxctx.logger.exception("Unexpected setup error")
-            res = 12
-        else:
-            # No exceptions while processing tar ball, success.
-            res = 0
-        finally:
-            if idxctx:
-                idxctx.dump_opctx()
-            idxctx.logger.debug("stopped processing list of tar balls")
-
-            idx = _count_lines(indexed)
-            skp = _count_lines(skipped)
-            err = _count_lines(erred)
-
-            idxctx.logger.info(
-                "{}.{}: indexed {:d} (skipped {:d}) results," " {:d} errors",
-                name,
-                idxctx.TS,
-                idx,
-                skp,
-                err,
-            )
-
-            if err > 0:
-                if skp > 0:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d}"
-                        f" results, w/ {err:d} errors"
-                    )
-                else:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, w/ {err:d}"
-                        " errors"
-                    )
-            else:
-                if skp > 0:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d} results"
-                else:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results"
+    idx = Index(name, options, idxctx, INCOMING_rp)
+    status, value = idx.collect_tb(ARCHIVE_rp, qdir)
 
-            report_fname = Path(tmpdir, f"{name}.{idxctx.TS}.report")
-            with report_fname.open(mode="w") as fp:
-                print(subj, file=fp)
-                if idx > 0:
-                    print("\nIndexed Results\n===============", file=fp)
-                    with indexed.open() as ifp:
-                        for line in sorted(ifp):
-                            print(line.strip(), file=fp)
-                if err > 0:
-                    print(
-                        "\nResults producing errors" "\n========================",
-                        file=fp,
-                    )
-                    with erred.open() as efp:
-                        for line in sorted(efp):
-                            print(line.strip(), file=fp)
-                if skp > 0:
-                    print("\nSkipped Results\n===============", file=fp)
-                    with skipped.open() as sfp:
-                        for line in sorted(sfp):
-                            print(line.strip(), file=fp)
-            try:
-                report.post_status(tstos(idxctx.time()), "status", report_fname)
-            except Exception:
-                pass
+    if not status:
+        return idx.process_tb(value)
 
-    return res
+    return value
```

Comment:
See above comment, perhaps should just be `return status`?

Reply:
The status is created by us on the basis that whether there are tarballs or not and the value is the return code

Reply:
Can you show in the code with a comment what contents of value will be when this return statement is executed?

Reply:
There can be 3  return codes 0, 9, and 12. We have already shown what each return code meant in the main function docstring. Should I show it again with a comment about three of them?

Reply:
```
except Exception:
    idxctx.logger.exception(
        "Unexpected error encountered generating list"
        " of tar balls to process"
    )   
    # tuple to return the status and return value
    return (1, 12) 
else:
    if not tarballs:
        idxctx.logger.info("No tar balls found that need processing")
        return (1, 0)

return (0, sorted(tarballs))
```

if I  proceed with this code. Here status is the first value and the second is the error code or tarball list. which can be dealt with like the following:
```
status, value = idx.collect_tb(ARCHIVE_rp, qdir)

if status == 0:
        value = idx.process_tb(value)

return value
```
if I rename value to tarballs then it doesn't make much sense as the return value.

if I move with a different approach:

```
except Exception:
    idxctx.logger.exception(
        "Unexpected error encountered generating list"
        " of tar balls to process"
    )   
    # tuple to return the status and return value
    return (1, 12) 
else:
    if not tarballs:
        idxctx.logger.info("No tar balls found that need processing")
        return (0, 0)

return (0, sorted(tarballs))
```

then I will have to check whether the tarballs list is empty or not and also the error code will be returned as 1 if the status conditions are not passed instead of 12
```
status, tarballs = idx.collect_tb(ARCHIVE_rp, qdir)

if status == 0:
    if len(tarballs):
        status = idx.process_tb(tarballs)

    return status
```
Am I missing something?
I believe reaching out to a better name should resolve this other than value and tarballs

Reply:
See the comments in `lib/pbench/server/indexing_tarballs.py`, if we change `collect_tb` to always return a status code and a list tuple, then we do the above code.  Also you don't need to do `if len(tarballs):`, you can achieve the same with `if tarballs:`, an empty list is `False`, and non-empty list is `True`.